src/main/java/com/minecolonies/blocks/BlockHutField.java

```diff
@@ -0,0 +1,239 @@
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * The class handling the fieldBlocks, placement and activation.
+ */
+public class BlockHutField extends BlockContainer
+{
+    /**
+     * Hardness of the block.
+     */
+    private static final float HARDNESS   = 10F;
+
+    /**
+     * Resistance of the block.
+     */
+    private static final float RESISTANCE = 10F;
+
+    /**
+     * The position it faces.
+     */
+    private static final PropertyDirection FACING = PropertyDirection.create("FACING", EnumFacing.Plane.HORIZONTAL);
+
+    /**
+     * Start of the collision box at y.
+     */
+    private static final double BOTTOM_COLLISION = 0.0;
+
+    /**
+     * Start of the collision box at x and z.
+     */
+    private static final double START_COLLISION = 0.1;
+
+    /**
+     * End of the collision box.
+     */
+    private static final double END_COLLISION = 0.9;
+
+    /**
+     * Height of the collision box.
+     */
+    private static final double HEIGHT_COLLISION = 2.5;
+
+    /**
+     * Constructor called on block placement.
+     */
+    BlockHutField()
+    {
+        super(Material.wood);
+        initBlock();
+    }
+
+    /**
+     * Getter of the description of the blockHut.
+     * @return the name as a String.
+     */
+    public String getName()
+    {
+        return "blockHutField";
+    }
+
+    /**
+     * Method called by constructor.
+     * Sets basic details of the block.
+     */
+    private void initBlock()
+    {
+        setRegistryName(getName());
+        setUnlocalizedName(Constants.MOD_ID.toLowerCase() + "." + getName());
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        //Blast resistance for creepers etc. makes them explosion proof.
+        setResistance(RESISTANCE);
+        //Hardness of 10 takes a long time to mine to not loose progress.
+        setHardness(HARDNESS);
+        this.setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH));
+        GameRegistry.registerBlock(this);
+        setBlockBounds((float)START_COLLISION, (float)BOTTOM_COLLISION, (float)START_COLLISION, (float)END_COLLISION, (float)HEIGHT_COLLISION, (float)END_COLLISION);
+    }
+
+    @Override
+    public boolean isFullCube()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isPassable(IBlockAccess worldIn, BlockPos pos)
+    {
+        return false;
+    }
+
+    @Override
+    public void onBlockPlacedBy(World worldIn, BlockPos pos, IBlockState state, EntityLivingBase placer, ItemStack stack)
+    {
+        //Only work on server side.
+        if(worldIn.isRemote)
+        {
+            return;
+        }
+
+        if(placer instanceof EntityPlayer)
+        {
+            Colony colony = ColonyManager.getColony(worldIn, pos);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'colony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)